SwanContents/swancontents/swanclassic/notebookapp.py

```diff
@@ -0,0 +1,52 @@
+# coding: utf-8
+
+from nbclassic.notebookapp import NotebookApp as ClassicNotebookApp
+
+from nbclassic import DEFAULT_TEMPLATE_PATH_LIST as NBCLASSIC_DEFAULT_TEMPLATE_PATH_LIST
+from nbclassic import DEFAULT_STATIC_FILES_PATH as NBCLASSIC_DEFAULT_STATIC_FILES_PATH
+
+from swancontents.swanclassic import DEFAULT_TEMPLATE_PATH_LIST, DEFAULT_STATIC_FILES_PATH
+from swancontents._version import __version__
+
+from jupyter_server.transutils import _i18n
+from jupyter_server.serverapp import load_handlers
+
+from traitlets import Unicode
+import datetime
+
+
+class NotebookApp(ClassicNotebookApp):
+
+    name = "swanclassic"
+    version = __version__
+    description = _i18n("""SWAN wrapper for nbclassic.
+
+    Serves the old Jupyter UI with the SWAN configurations (i.e templates) on top""")
+
+    default_url = Unicode("/projects").tag(config=True)
+
+    static_paths = [DEFAULT_STATIC_FILES_PATH, NBCLASSIC_DEFAULT_STATIC_FILES_PATH]
+    template_paths = DEFAULT_TEMPLATE_PATH_LIST + NBCLASSIC_DEFAULT_TEMPLATE_PATH_LIST
+
+    def initialize_handlers(self):
+        handlers = []
+        handlers.extend(load_handlers('swancontents.swanclassic.handlers.projects'))
+        handlers.extend(load_handlers('swancontents.swanclassic.handlers.share'))
+        handlers.extend(load_handlers('swancontents.swanclassic.handlers.cernbox'))
+        self.handlers.extend(handlers)
+        super(NotebookApp, self).initialize_handlers()
+
+    def initialize_settings(self):
+        super(NotebookApp, self).initialize_settings()
+        new_vars = self.settings['jinja_template_vars'] if 'jinja_template_vars' in self.settings else dict()
+        new_vars.update({'current_year':datetime.datetime.now().year})
+
+        if datetime.date.today().month == 12:
+            # It's Christmas time!
```

Comment:
:D How is this here? The SWANta only shows in the main page that is served by the hub, before the notebook server is even started, correct?

Reply:
This includes SWANta in the download page, which also shows the same loading effect as the hub.